rust/template/src/api/c_api.rs

```diff
@@ -35,45 +35,88 @@ use std::os::unix::io::{FromRawFd, IntoRawFd, RawFd};
 use std::os::windows::io::{FromRawHandle, IntoRawHandle, RawHandle};
 
 #[no_mangle]
-pub unsafe extern "C" fn ddlog_get_table_id(tname: *const raw::c_char) -> libc::size_t {
-    if tname.is_null() {
+pub unsafe extern "C" fn ddlog_get_table_id(
+    prog: *const HDDlog,
```

Comment:
If you want to be clear about the contract that functions expect, you can use `NonNull<HDDlog>` to signify non-null pointers and `Option<NonNull<T>>` for nullable ones. This can be done in both parameter and return positions and allows the backing rust code niceties like `?`

Reply:
It's undefined behavior, same as what happens with the current code. If nulls are expected (which I don't see why why would be) then `Option<NonNull<T>>` does the same thing with the ability to take a null value as well